File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if(head == NULL || head->next == NULL){
            return NULL;
        }
        int cnt = 1;
        ListNode *curr = head;
        while(curr->next!=NULL){
            curr = curr->next;
            cnt ++;
        }
        if(n==cnt){
            return head->next;
        }
        ListNode *temp = head;
        int i = 0;
        while( i < cnt-n-1){
            temp = temp->next;
            i++;
        }
        temp->next = temp->next->next;
        return head;
    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        if(head==null || head.next==null){
            return null;
        }
        ListNode curr=head;
        int size=1;
        while(curr.next!=null){
            curr=curr.next;
            size++;
        }
        if(n==size){
            return head.next;
        }
        int ptf=size-n;
        int st=0;
        ListNode temp=head;
        while(st<ptf-1){
            temp=temp.next;
            st++;
        }
        temp.next=temp.next.next;
        return head;
    }
}